File: src/ois_valentinesday/valentinesday.pas
{ NOTE: it is recommended to use this even if you don't understand the following code }

const
    MAXN = 100000;

var
    M, N, i, maxChocolates : LongInt;
    C, P                   : Array[0..MAXN-1] of LongInt;

begin
{
    uncomment the two following lines if you want to read/write from files
    assign(input,  'input.txt');  reset(input);
    assign(output, 'output.txt'); rewrite(output);
}

    ReadLn(N, M);

    for i:=0 to N-1 do
        Read(C[i]);
    ReadLn();

    for i:=0 to N-1 do
        Read(P[i]);
    ReadLn();

    maxChocolates := 0;

    { INSERT YOUR CODE HERE }


    WriteLn(maxChocolates);

end.

File: src/ois_valentinesday/valentinesday.c
// NOTE: it is recommended to use this even if you don't understand the following code.

#include <assert.h>
#include <stdio.h>
#include <stdlib.h>

#define MAXN 100000

int M, N, i, maxChocolates;
int C[MAXN], P[MAXN];

int main() {
    // uncomment the two following lines if you want to read/write from files
    // freopen("input.txt", "r", stdin);
    // freopen("output.txt", "w", stdout);

    assert(2 == scanf("%d%d", &N, &M));

    for (i = 0; i < N; ++i)
        assert(1 == scanf("%d", &C[i]));

    for (i = 0; i < N; ++i)
        assert(1 == scanf("%d", &P[i]));

    maxChocolates = 0;


    // INSERT YOUR CODE HERE


    printf("%d\n", maxChocolates);

    return 0;
}
